libpromises/evalfunction.c

```diff
@@ -3039,6 +3039,42 @@ static FnCallResult FnCallReadTcp(ARG_UNUSED EvalContext *ctx,
 
 /*********************************************************************/
 
+static FnCallResult FnCallIsConnectable(ARG_UNUSED EvalContext *ctx,
+                                        ARG_UNUSED const Policy *policy,
+                                        const FnCall *fp,
+                                        const Rlist *finalargs)
+{
+    assert(fp != NULL);
+    if (finalargs == NULL)
+    {
+        Log(LOG_LEVEL_ERR, "Function %s requires a variable as first argument",
+            fp->name);
+        return FnFailure();
+    }
+    char *hostnameip = RlistScalarValue(finalargs);
+    if (finalargs->next == NULL)
+    {
+        Log(LOG_LEVEL_ERR, "Function %s requires a variable as second argument",
+            fp->name);
+        return FnFailure();
+    }
+    char *port = RlistScalarValue(finalargs->next);
+    if (finalargs->next->next == NULL)
```

Comment:
The third argument could be optional, defaulting to 100 ms for example

Reply:
https://docs.cfengine.com/docs/3.24/reference-functions-isreadable.html also has the timeout as an optional parameter